pkg/controller/upload_controller.go

```diff
@@ -327,6 +327,27 @@ func (c *uploadController) processUpload(req *pluginv1api.Upload) error {
 		return errors.New(errMsg)
 	}
 
+	// If upload phase is UploadPhaseCleanupFailed, manually call DeleteLocalSnapshot to retry on deleting local snapshot
+	if req.Status.Phase == pluginv1api.UploadPhaseCleanupFailed {
+		err = c.snapMgr.DeleteLocalSnapshot(peID)
+		if err != nil {
+			errMsg := fmt.Sprintf("Failed to clean up local snapshot, %v, will retry. %v", peID.String(), errors.WithStack(err))
+			_, err = c.patchUploadByStatusWithRetry(req, pluginv1api.UploadPhaseCleanupFailed, errMsg)
+			if err != nil {
+				errMsg = fmt.Sprintf("%v. %v", errMsg, errors.WithStack(err))
```

Comment:
I think here we should not log the same "errMsg" again.  Instead this message should be rewritten to indicate patchUploadByStatusWithRetry failed.